pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -175,18 +175,37 @@ func toMachineTemplate(machinePoolName string, cluster *rancherv1.Cluster, machi
 		machinePoolData.SetNested(secretName, "common", "cloudCredentialSecretName")
 	}
 
+	annotations := map[string]any{
+		rke2.MachineTemplateClonedFromGroupVersionAnn: gvk.GroupVersion().String(),
+		rke2.MachineTemplateClonedFromKindAnn:         gvk.Kind,
+		rke2.MachineTemplateClonedFromNameAnn:         machinePool.NodeConfig.Name,
+	}
+
+	hostnameLimit := 0
+	if hl := machinePool.HostnameLengthLimit; hl >= rke2.MinimumHostnameLengthLimit && hl <= rke2.MaximumHostnameLengthLimit {
+		hostnameLimit = hl
+	} else if hl = cluster.Spec.RKEConfig.MachinePoolDefaults.HostnameLengthLimit; hl >= rke2.MinimumHostnameLengthLimit && hl <= rke2.MaximumHostnameLengthLimit {
+		hostnameLimit = hl
+	}
+
+	if hostnameLimit != 0 {
+		annotations[rke2.HostnameLengthLimitAnnotation] = machinePool.HostnameLengthLimit
+	}
+
+	if machinePool.HostnameLengthLimit < rke2.MinimumHostnameLengthLimit {
```

Comment:
Do we also need to add a debug message and not set the annotation if `machinePool.HostnameLengthLimit` > `rke2.MinimumHostnameLengthLimit` ?

Reply:
I'm going to add tests for this to ensure the hostname limit annotation is set correctly.